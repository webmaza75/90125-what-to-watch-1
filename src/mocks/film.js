const film = {
  genre: [`Drama`, `Thriller`],
  title: `Aviator`,
  desc: ``,
  picture: `aviator.jpg`,
  year: 2016,
  src: ``
};

export default film;
