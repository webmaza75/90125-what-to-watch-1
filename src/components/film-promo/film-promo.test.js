import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import FilmPromo from './film-promo.jsx';
import film from '../../mocks/film.js';

it(`FilmPromo correctly renders`, () => {
  const {
    posterImage,
    title,
    genre,
    released,
    backgroundImage,
    id,
    isFavorite
  } = film;

  const renderer = new ShallowRenderer();
  const result = renderer
    .render(<FilmPromo
      posterImage={posterImage}
      title={title}
      genre={genre}
      released={released}
      backgroundImage={backgroundImage}
      id={id}
      isFavorite={isFavorite}
    />);

  expect(result).toMatchSnapshot();
});
