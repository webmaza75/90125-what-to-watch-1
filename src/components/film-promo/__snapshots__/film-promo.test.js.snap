// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmPromo correctly renders 1`] = `
<React.Fragment>
  <div
    className="movie-card__bg"
  >
    <img
      alt="War of the Worlds"
      src="https://es31-server.appspot.com/wtw/static/film/background/War_of_the_Worlds.jpg"
    />
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <Memo(ConnectFunction) />
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="War of the Worlds poster"
          height="327"
          src="https://es31-server.appspot.com/wtw/static/film/poster/War_of_the_Worlds.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          War of the Worlds
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Adventure
          </span>
          <span
            className="movie-card__year"
          >
            2005
          </span>
        </p>
        <Memo(ConnectFunction)
          id={1}
          isFavorite={false}
          showAddReviewLink={false}
        />
      </div>
    </div>
  </div>
</React.Fragment>
`;
